07-On demand scan as an API/README.md

```diff
@@ -0,0 +1,203 @@
+- Using user authentication code flow
+
+#### Required Az Module
+``` PowerShell
+Install-Module -Name MSAL.PS -AllowClobber -Scope CurrentUser -repository PSGallery
+```
+
+### Using client credential flow
+Client crediential flow uses the client credentials(client id and client secret) to generate the token. Token will be generating against specified SPN(Service Principal Name) and **SPN must have access over the subscription** to scan or to get the control scan result.
+
+> In order to generate the token for APIs, you have to get access for the client application from WebAPI owner.
+> 1. Send the client id to WebAPI owner to grant access for client application.
+> 2. WebAPI owner will grant the access and share the scope.
+> 3. Use WebAPI scope in below command.
+
+Commands to generate the token:
+``` PowerShell
+
+$ClientSecret = '<client-secret>' | ConvertTo-SecureString -AsPlainText -Force
+
+$token = Get-MsalToken -TenantId '<tenant-id>' -ClientId '<client-id>' -ClientSecret $ClientSecret -Scopes "<WebAPI-scope>/.default"
+
+$token.AccessToken | ConvertTo-Json | Out-File '<token file path>\token.json'
+
+```
+
+[Back to top…](README.md#On-this-page)
+
+### Using user authentication code flow
+User authentication code flow uses user's crediential to generate the token. User must have access over the subscription  to scan or to get the control scan result.
+
+> In this flow, Client app registration needs "Admin consent" to access the APIs.
+> </br>
+> Grant admin consent for client app registration:
+> 1. Go to Azure Portal.
+> 2. Go to App Registration.
+> 3. Select appropriate App Registration.
+> 4. Go to 'API Permissions'.
+> 5. Click 'Grant admin consent for Microsoft'.
+
+
+Command to generate the token:
+``` PowerShell
+
+$token = Get-MsalToken -TenantId '<tenant-id>' -ClientId '<client-app-id>' -RedirectUri 'https://localhost' -Scopes '<WebAPI-scope>'
+
+```
+
+[Back to top…](README.md#On-this-page)
+
+## API to scan a subscription
+
+Use below PowerShell command to get control scan result.
+
+``` PowerShell
+
+$header = "Bearer " + $token.AccessToken
+$headers = @{"Authorization"=$header;"Content-Type"="application/json";}
+$requestBody = @{"SubscriptionIDList"=@("subscription_id");}
+
+# Example:
+$requestBody = @{"SubscriptionIDList"=@("sub1","sub2");}
+
+
+$method = [Microsoft.PowerShell.Commands.WebRequestMethod]::POST
+$apiUri = "https://azsk-ats-api-ringn.azurewebsites.net/adhocscan/RequestScan"
+
+$apiResponse = Invoke-WebRequest -Method $method -Uri $ascUri -Headers $headers -Body ($requestBody | ConvertTo-Json) -UseBasicParsing
+
+$response = ConvertFrom-Json $apiResponse.Content
+$response
+
+```
+
+**Request body parameter details:**
+|Param Name|Description|Required?
+|----|----|----|
+| SubscriptionId| List of subscription id(s) for scan. | Yes|
+
+</br>
+
+[Back to top…](README.md#On-this-page)
+
+## API to get control scan result
+Use below PowerShell command to get control scan result.
+
```

Comment:
Same as above. Description + Data contract

Reply:
Added description + important points as part of data contract.